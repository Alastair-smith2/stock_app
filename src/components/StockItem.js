import React from "react";
import { View, Text, StyleSheet, Platform } from "react-native";

const StockItem = ({ item }) => {
  return (
    <View style={styles.container}>
      <View style={styles.date}>
        <Text style={styles.text}>Date - {item.date}</Text>
      </View>
      <View style={styles.padding} />
      <View style={styles.high}>
        <Text style={styles.text}>High- {item.data["2. high"]}</Text>
      </View>
      <View style={styles.padding} />
      <View style={styles.low}>
        <Text style={styles.text}>Low - {item.data["3. low"]}</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 0.2,
    flexDirection: "row",
    marginBottom: 10,
    justifyContent: "center"
  },
  date: {
    flex: 0.25
  },
  padding: {
    flex: 0.1
  },
  high: {
    flex: 0.25
  },
  low: {
    flex: 0.25
  },
  text: {
    textAlign: "center",
    ...Platform.select({
      ios: {
        color: "black"
      },
      android: {
        color: "#212121"
      }
    })
  }
});
export default StockItem;
