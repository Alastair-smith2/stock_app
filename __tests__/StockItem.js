import React from "react";
import { shallow } from "enzyme";

import StockItem from "../src/components/StockItem.js";

describe("The StockItem component", () => {
  let wrapper;
  beforeEach(() => {
    let item = {
      data: {
        date: "2017-06-10",
        "2. high": 196,
        "3. low": 100
      }
    };
    wrapper = shallow(<StockItem item={item} />);
  });
  describe("It should render", () => {
    it("Should render", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
