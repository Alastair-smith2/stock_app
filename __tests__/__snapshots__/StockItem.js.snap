// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The StockItem component It should render Should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StockItem
    item={
      Object {
        "data": Object {
          "2. high": 196,
          "3. low": 100,
          "date": "2017-06-10",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "flex": 0.25,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "textAlign": "center",
              }
            }
          >
            Date - 
          </Text>
        </View>,
        <View
          style={
            Object {
              "flex": 0.1,
            }
          }
        />,
        <View
          style={
            Object {
              "flex": 0.25,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "textAlign": "center",
              }
            }
          >
            High- 
            196
          </Text>
        </View>,
        <View
          style={
            Object {
              "flex": 0.1,
            }
          }
        />,
        <View
          style={
            Object {
              "flex": 0.25,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "textAlign": "center",
              }
            }
          >
            Low - 
            100
          </Text>
        </View>,
      ],
      "style": Object {
        "flex": 0.2,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 10,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "textAlign": "center",
              }
            }
          >
            Date - 
          </Text>,
          "style": Object {
            "flex": 0.25,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": Array [
              "Date - ",
              undefined,
            ],
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "black",
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            "Date - ",
            undefined,
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "flex": 0.1,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "textAlign": "center",
              }
            }
          >
            High- 
            196
          </Text>,
          "style": Object {
            "flex": 0.25,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": Array [
              "High- ",
              196,
            ],
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "black",
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            "High- ",
            196,
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "flex": 0.1,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "textAlign": "center",
              }
            }
          >
            Low - 
            100
          </Text>,
          "style": Object {
            "flex": 0.25,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": Array [
              "Low - ",
              100,
            ],
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "black",
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            "Low - ",
            100,
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "flex": 0.25,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "black",
                  "textAlign": "center",
                }
              }
            >
              Date - 
            </Text>
          </View>,
          <View
            style={
              Object {
                "flex": 0.1,
              }
            }
          />,
          <View
            style={
              Object {
                "flex": 0.25,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "black",
                  "textAlign": "center",
                }
              }
            >
              High- 
              196
            </Text>
          </View>,
          <View
            style={
              Object {
                "flex": 0.1,
              }
            }
          />,
          <View
            style={
              Object {
                "flex": 0.25,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "black",
                  "textAlign": "center",
                }
              }
            >
              Low - 
              100
            </Text>
          </View>,
        ],
        "style": Object {
          "flex": 0.2,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "black",
                  "textAlign": "center",
                }
              }
            >
              Date - 
            </Text>,
            "style": Object {
              "flex": 0.25,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                "Date - ",
                undefined,
              ],
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "black",
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": Array [
              "Date - ",
              undefined,
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "flex": 0.1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "black",
                  "textAlign": "center",
                }
              }
            >
              High- 
              196
            </Text>,
            "style": Object {
              "flex": 0.25,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                "High- ",
                196,
              ],
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "black",
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": Array [
              "High- ",
              196,
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "flex": 0.1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "black",
                  "textAlign": "center",
                }
              }
            >
              Low - 
              100
            </Text>,
            "style": Object {
              "flex": 0.25,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                "Low - ",
                100,
              ],
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "black",
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": Array [
              "Low - ",
              100,
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
